lib/membrane_ffmpeg_swresample_plugin/converter.ex

```diff
@@ -234,18 +237,43 @@ defmodule Membrane.FFmpeg.SWResample.Converter do
     end
   end
 
-  defp update_pts_queue(state, converted_frames_count) do
-    case state.pts_queue do
-      [{out_pts, expected_frames} | rest] ->
-        if converted_frames_count < expected_frames do
-          {%{state | pts_queue: [{out_pts, expected_frames - converted_frames_count}] ++ rest},
-           out_pts}
-        else
-          {%{state | pts_queue: rest}, out_pts}
-        end
+  defp update_pts_queue(state, converted_frames) do
+    [{out_pts, expected_frames} | rest] = state.pts_queue
+
+    cond do
+      converted_frames < expected_frames ->
+        {%{state | pts_queue: [{out_pts, expected_frames - converted_frames}] ++ rest}, out_pts}
+
+      converted_frames > expected_frames ->
+        remaining_frames_count = converted_frames - expected_frames
+
+        filtered = filter_pts_queue(rest, remaining_frames_count)
+
+        {%{state | pts_queue: filtered}, out_pts}
```

Comment:
isn't it the same as 
```suggestion
        update_pts_queue(rest, remaining_frames_count)
```